pkg/solidity-utils/contracts/math/BasePoolMath.sol

```diff
@@ -206,9 +198,8 @@ library BasePoolMath {
         uint256 amountIn = newBalance - currentBalances[tokenInIndex];
 
         // Calculate the non-taxable amount, which is the new balance proportionate to the BPT minted.
-        // Round the `nonTaxableBalance` down to favor the protocol by increasing the taxable amount, which charges
-        // higher swap fees, ultimately increasing the amount of `tokenIn` that will be transferred from the caller.
-        uint256 nonTaxableBalance = newSupply.mulDown(currentBalances[tokenInIndex]).divDown(totalSupply);
+        // Since we multiply and divide we don't need to use FP math.
+        uint256 nonTaxableBalance = (newSupply * currentBalances[tokenInIndex]) / totalSupply;
```

Comment:
I believe we're using BasePoolMath only for scaled18 values, but it's worth noting that `currentBalances[tokenInIndex]` must be an 18 decimals number.

Reply:
I think that was the case already right? FP math amplifies or divides by 1e18, and that only works if we're multiplying 18 decimal values.